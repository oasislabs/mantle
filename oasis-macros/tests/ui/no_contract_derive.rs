oasis_std::contract! {

pub struct Counter(u32);

impl Counter {
    pub fn new(ctx: &Context) -> Self {
        Self(42)
    }
}

}

fn main() {}
