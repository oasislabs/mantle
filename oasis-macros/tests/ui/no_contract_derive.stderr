error: Contract definition must contain a #[derive(Contract)] struct.
  --> $DIR/no_contract_derive.rs:1:1
   |
1  | / oasis_std::contract! {
2  | |
3  | | pub struct Counter(u32);
4  | |
...  |
10 | |
11 | | }
   | |_^

error: aborting due to previous error

